server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java

```diff
@@ -22,138 +24,425 @@
+                allowEmptyExpressions = options.allowEmptyExpressions;
+            }
+
+            /**
+             * Open indices will be matched. Defaults to true.
+             */
+            public Builder matchOpen(boolean matchOpen) {
+                this.matchOpen = matchOpen;
+                return this;
+            }
+
+            /**
+             * Closed indices will be matched. Default to false.
+             */
+            public Builder matchClosed(boolean matchClosed) {
+                this.matchClosed = matchClosed;
+                return this;
+            }
+
+            /**
+             * Hidden indices will be included from the result. Defaults to false.
+             */
+            public Builder includeHidden(boolean includeHidden) {
+                this.includeHidden = includeHidden;
+                return this;
+            }
+
+            /**
+             * Aliases will be included in the result. Defaults to true.
+             */
+            public Builder resolveAliases(boolean resolveAliases) {
+                this.resolveAliases = resolveAliases;
+                return this;
+            }
+
+            /**
+             * If true, when any of the expressions does not match any indices, we consider the result of this expression
+             * empty; if all the expressions are empty then we have a successful but empty response.
+             * If false, we throw an error immediately, so even if other expressions would result into indices the response
+             * will contain only the error. Defaults to true.
+             */
+            public Builder allowEmptyExpressions(boolean allowEmptyExpressions) {
+                this.allowEmptyExpressions = allowEmptyExpressions;
+                return this;
+            }
+
+            /**
+             * Disables expanding wildcards.
+             */
+            public Builder none() {
+                matchOpen = false;
+                matchClosed = false;
+                includeHidden = false;
+                return this;
+            }
+
+            /**
+             * Maximises the resolution of indices, we will match open, closed and hidden targets.
+             */
+            public Builder all() {
+                matchOpen = true;
+                matchClosed = true;
+                includeHidden = true;
+                return this;
+            }
+
+            /**
+             * Parses the list of wildcard states to expand as provided by the user.
+             * Logs a warning when the option 'none' is used along with other options because the position in the list
+             * changes the outcome.
+             */
+            public Builder expandStates(String[] expandStates) {
+                // Calling none() simulates a user providing an empty set of states
+                none();
+                for (String expandState : expandStates) {
+                    switch (expandState) {
```

Comment:
Can we add a `default` branch that throws an exception so we don't silently ignore an option that was added but not handled here?